### PR TITLE
[Analytics] Fix path prop of dashboard_clicked event

### PR DESCRIPTION
**Pull request by jakobhero**
## Description

This PR fixes an Analytics bug in the tracking of clicks in the dashboard where the path an anchor redirects to is passed to `path` of the `dashboard_clicked` track call instead of the path from which the anchor was clicked by moving the activation of the event handler to the capturing phase.

## How to test

- open `/project` in the preview environment
- click on Branches or Prebuild of any project
- check in Segment's staging untrusted source that the path for the fired `dashboard_clicked` is `/project`, not `/project/<name_of_project_the_link_redirects_to>[/prebuilds]`

## Release Notes

```release-note
NONE
```

## Documentation

/werft analytics=segment|TEZnsG4QbLSxLfHfNieLYGF4cDwyFWoe

### CODE REVIEW

**Comment by roboquat** [comment]
@jakobhero: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **laushinka** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @laushinka` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jankeromnes — APPROVED** [review]
Looks good to me, thanks! 🎉 

Quite a subtle bug you solved there 😅 (today I learned about the "capturing" vs "bubbling" phases! I imagine it's all the Single-Page-App fast `<Link to/>` and `history.push()` that change the `window.location.pathname` dynamically prior to the event "bubbling" back.)

Anyway, I'm a fan 🤓 🙏 

/approve no-issue
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: f46b638189f1dfed9f4e5f1d994e8f653a112674</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6917#pullrequestreview-817118491" title="Approved">jankeromnes</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6917#pullrequestreview-817118491" title="Approved">jankeromnes</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
